src/gl/canvas2d.re

```diff
@@ -0,0 +1,76 @@
+type ctx;
+
+/* Canvas API, following https://simon.html5.org/dump/html5-canvas-cheat-sheet.html */
+
+external save : ctx => unit = "save" [@@bs.send];
+external restore : ctx => unit = "restore" [@@bs.send];
+
+/* Transformation */
+external scale : ctx => float => float => unit = "scale" [@@bs.send];
```

Comment:
Could you convert the `send` externals to `send.pipe`? E.g. `external scale : float => float => unit = "" [@@bs.send.pipe: ctx];`. It will put the `ctx` argument at the end instead, so you can then directly use it with the pipe operator (`|>`) and more easily apply it partially. This is the more idiomatic form.